sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -26,19 +26,36 @@ object ScalaNativePluginInternal {
   val nativeWorkdir =
     taskKey[File]("Working directory for intermediate build files.")
 
+  private val nativeStandardLibraries =
+    Seq("nativelib", "clib", "posixlib", "windowslib", "javalib", "auxlib")
+
   lazy val scalaNativeDependencySettings: Seq[Setting[_]] = Seq(
     libraryDependencies ++= Seq(
-      "org.scala-native" %%% "nativelib" % nativeVersion,
-      "org.scala-native" %%% "javalib" % nativeVersion,
-      "org.scala-native" %%% "auxlib" % nativeVersion,
       "org.scala-native" %%% "test-interface" % nativeVersion % Test
     ),
     libraryDependencies += CrossVersion
       .partialVersion(scalaVersion.value)
       .fold(throw new RuntimeException("Unsupported Scala Version")) {
+        // Add only dependency to scalalib, nativeStanardLibraries would be added transitively
         case (2, _) => "org.scala-native" %%% "scalalib" % nativeVersion
         case (3, _) => "org.scala-native" %%% "scala3lib" % nativeVersion
       },
+    excludeDependencies ++= {
+      // Exclude cross published version dependencies leading to conflicts in Scala 3 vs 2.13
+      // When using Scala 3 exclude Scala 2.13 standard native libraries,
+      // when using Scala 2.13 exclude Scala 3 standard native libraries
+      // Use full name, Maven style published artifacts cannot use artifact/cross version for exclusion rules
+      nativeStandardLibraries.map { lib =>
+        val scalaBinVersion =
+          if (scalaVersion.value.startsWith("3.")) "2.13"
+          else "3"
+        ExclusionRule()
+          .withOrganization("org.scala-native")
+          .withName(
+            s"${lib}_native${ScalaNativeCrossVersion.currentBinaryVersion}_${scalaBinVersion}"
+          )
+      }
+    },
```

Comment:
@lolgab You're way more experienced with mill build tool, do you think we would be able to apply this solution in mill?

@jchyb You worked with scala-cli, do you think it would possible to apply this fix there as well?

Reply:
@WojciechMazur Yes, I think it is possible. I'll try to make a PR to implement this logic and let you know :)

Reply:
@WojciechMazur it should be possible in scala-cli as well, I'll try making a PR next week (hopefully before scala-cli's next release)

Reply:
Thank you! According to the PR's let's keep them as the drafts until this we would be sure that we can use this workaround everywhere

Reply:
@WojciechMazur This should be it :)
https://github.com/com-lihaoyi/mill/pull/1738